SocialVideoHelper.m

```diff
@@ -236,32 +259,95 @@ +(void)tweetVideoStage3:(NSData*)videoData mediaID:(NSString *)mediaID comment:(
+            NSLog(@"Twitter Error stage 4 - %@", error);
+            NSLog(@"Twitter Error stage 4 - %@", urlResponse);
+            [SocialVideoHelper uploadError:error withCompletion:completion];
+            return;
+        } else {
+            if ([urlResponse statusCode] == 200){
+//                NSLog(@"Twitter STATUS upload success !");
+                
+                NSMutableDictionary *returnedData = [NSJSONSerialization JSONObjectWithData:responseData options:NSJSONReadingMutableContainers error:&error];
+                NSDictionary *processing_infor = [returnedData valueForKey:@"processing_info"];
+                
+                NSString *state = [NSString stringWithFormat:@"%@", [processing_infor valueForKey:@"state"]];
+                NSString *check_after_secs = [NSString stringWithFormat:@"%@", [processing_infor valueForKey:@"check_after_secs"]];
+                NSString *progress_percent = [NSString stringWithFormat:@"%@", [processing_infor valueForKey:@"progress_percent"]];
+                NSLog(@"state : %@, check_after_secs: %@, progress_percent: %@\n",state, check_after_secs, progress_percent);
+                
+                if ([state  isEqual: @"in_progress"]) {
+                    NSLog(@"%@",[NSThread currentThread]);
+                    sleep([check_after_secs intValue]);
+                    [SocialVideoHelper tweetVideoStage4:videoData mediaID:mediaID comment:comment account:account withCompletion:completion];
+                } else if ([state  isEqual: @"pending"]){
+                    NSLog(@"%@",[NSThread currentThread]);
+                    sleep([check_after_secs intValue]);
+                    [SocialVideoHelper tweetVideoStage4:videoData mediaID:mediaID comment:comment account:account withCompletion:completion];
+                } else if ([state  isEqual: @"failed"]) {
+                    NSLog(@"tweetVideoStage4 return state : failed");
+                    return; //TODO HUD
+                } else if ([state  isEqual: @"succeeded"]) {
+                    NSLog(@"stage four success, mediaID -> %@", mediaID);
+                    [SocialVideoHelper tweetVideoStage5:videoData mediaID:mediaID comment:comment account:account withCompletion:completion];
+                }
+            }
+        }
+    }];
+    
+}
+
++(void)tweetVideoStage5:(NSData*)videoData mediaID:(NSString *)mediaID comment:(NSString*)comment account:(ACAccount*)account withCompletion:(VideoUploadCompletion)completion{
     NSURL *twitterPostURL = [[NSURL alloc] initWithString:@"https://api.twitter.com/1.1/statuses/update.json"];
     
     if (comment == nil) {
-        comment = [NSString stringWithFormat:@"#SocialVideoHelper# https://github.com/liu044100/SocialVideoHelper"];
+    // try to not prefill the post status text for user
+//        comment = [NSString stringWithFormat:@"#SocialVideoHelper# https://github.com/liu044100/SocialVideoHelper"];
+        comment = [NSString stringWithFormat:@"  "];
+
     }
```

Comment:
Sorry about that, but I want to retain `comment = [NSString stringWithFormat:@"#SocialVideoHelper# https://github.com/liu044100/SocialVideoHelper"];`, because sometimes search this comment in twitter is interesting.